### ISSUE TITLE
Login information sporadically forgotten

### ISSUE THREAD

**Issue by setyb** [bitwarden/browser#544]
About once every 3-4 weeks, bitwarden will "forget" the username and master password for the bitwarden account.

I wonder if this is a timing issue, race condition, or Firefox bug.  It's like all the bitwarden local data is deleted when either Firefox is (a) opened or (b) closed.  None of the data within the vault is affected.

Test system: bitwarden v1.24.0, Firefox 59.0, Win7

**Comment by kspearrin** [comment]
Sorry,not sure i understand the issue. What do you mean by "Forgot" your username/password? Are you saying that you get logged out or something?

**Comment by setyb** [comment]
Exactly.  Bitwarden acts like the first time it is installed.  It forces me to re-enter the username and password to access the vault.

**Comment by kspearrin** [comment]
Do you have some addon or make some change that would clear your firefox user profile?

**Comment by setyb** [comment]
Good question.  The answer is that nothing is clearing the profile itself.  I do, however, have Firefox set to clear everything (cookies/history/cache/offline website data/etc.) when it terminates.

After closing Firefox, and then restarting it, everything starts back up fine, including all other extensions.  And bitwarden loads fine, it just forgets my login information once in a while.

The first time it happened, I figured it was just a fluke.  But it is consistently happening every few weeks. so it appears to be more than a fluke. :)

All bitwarden settings are also lost when this happens, including custom preferences set in bitwarden's **Settings -> Options**.

**Comment by setyb** [comment]
That's the interesting thing.  None of the other extensions are losing their data.  It's only bitwarden.

**Comment by arethsu** [comment]
I have experienced this as well. Every few weeks it logs me out randomly. For reference, I always run the latest Firefox Nightly, and I am a Yubikey user.

I was thinking that it had something to do with the Yubikey token expiring every now and then, but it feels like the "reset"/logout happens randomly.

**Comment by kspearrin** [comment]
Are you guys using the "remember me" 2fa option by chance?

**Comment by setyb** [comment]
I am not.

**Comment by arethsu** [comment]
I am using it, which is why I at first assumed it was some kind of regular interval expiry (which is totally fine, it should be that way). The timing/interval of the "logout" feels random though. Maybe I'm delusional. :/

**Comment by arethsu** [comment]
Hmm, it seems that I got logged out again, this time, I think, after updating Firefox to the latest nightly. I still don't know if this is the cause. I will continue to observe.

**Comment by Crocmagnon** [comment]
Would it be possible that you're logged out when FF updates or when your profile is reset/modified ?

**Comment by setyb** [comment]
@Crocmagnon I had the same thought.  It's definitely not *every* FF update.  But I'm now keeping a log of when it happens to see if there is any correlation.

**Comment by Crocmagnon** [comment]
Another idea : I'm personally logged out after an update of the bitwarden extension, would that be plausible ?

**Comment by kspearrin** [comment]
I don't know. I have never experienced this on Firefox stable so it's hard to say.

**Comment by setyb** [comment]
OK... It happened again today.  I notice that I created this issue on March 9, and today is April 9.  So exactly 1 month.  Unlikely a coincidence.

No Firefox or Bitwarden extension updates today, so that's not it.

**Comment by Crocmagnon** [comment]
30 days seems like a timeout for the 2fa, I think I've seen this somewhere else...

**Comment by kspearrin** [comment]
Yes, a 2fa remember me session will last 30 days, but 2fa should not be part of any interaction with the APIs once you initially log in. After that it is all JWT and refresh tokens interacting with the identity server.

**Comment by setyb** [comment]
Happened again today, this time right after performing a Firefox update from 60.0b11 to 60.0b12.

**Comment by setyb** [comment]
Happened again today using BW 1.26.5 on FF60.0.

Seems to be about once a month.

No OS, Firefox, or BW updates today.

Opened browser and it worked fine.  Closed browser.  Opened browser again, and I was suddenly logged out of BW.

**Comment by setyb** [comment]
Happened again today using BW 1.28.0 on FF62.0beta12.

It logged in successfully about 20 minutes ago.  Then I rebooted the machine.  Upon rebooting and opening the browser, bitwarden had forgotten the login info.

No OS, Firefox, or BW update today.  Did have another extension with an update today.

**Comment by setyb** [comment]
@kspearrin I came across [this issue](https://github.com/stoically/temporary-containers/issues/68) for a different Firefox extension.  Is it possible that this bitwarden issue is caused by the same underlying issue as that one?

**Comment by kspearrin** [comment]
@setyb I dont think so. We don't ever do anything to wipe storage.

**Comment by setyb** [comment]
@kspearrin Thanks.  It was worth a try.

BTW, one of the side-effects of this issue is that you lose all the MRU ordering of logins for sites with multiple logins.

**Comment by setyb** [comment]
The issue hasn't happened in months, but then just happened.  Now with FF 66.0.3 (64-bit).  Nothing updated recently.

**Comment by urholaukkarinen** [comment]
This has happened to me many times over the last year or so. It had worked fine for several weeks, but today it happened again. I have a dual boot system with Linux and Windows. Yesterday I logged in to the Bitwarden Firefox plugin on Linux and today I noticed that the same plugin on Windows had forgotten login information. Could this be somehow related to Firefox sync?

**Comment by cwmke** [comment]
I have two family members experiencing this with Brave so I'm not sure it's specific to Firefox. Every few weeks their Bitwarden extension logs out for no apparent reason. It's causing some frustration on their end so any guidance is appreciated.

**Comment by nelwa** [comment]
I can confirm that I'm experiencing the same problem.  Not only in Firefox though, also on Chromium.  I use Linux on my systems, and I experience the same problem on my laptop as well as my pc - though not at the same time on both.  It's also a system wide issue - when it happens, it happens on all browsers on that system.  It is very annoying because my master password is long & complicated, and to type it into multiple browsers each time this happens is very frustrating.  Other than this issue, bitwarden is perfect, and I recommend it to everyone.

I have not yet been able to identify a specific cause - I can rule out the following: system updates, browser updates, browser plugin updates, any type of sign in on other devices.  

It seems to be some sort of random timeout (approx 3-4 weeks) whether I'm using the system or not.

**Comment by MarcusWolschon** [comment]
You're not alone.
Same issue here.
No clue as to what causes it.

**Comment by cwmke** [comment]
Not for me. It was happening more frequently then 30 days. I haven’t had any complaints in quite a while so I’m assuming it’s good now?

**Comment by closebot-bw** [comment]
⚠️ **Stale Issue Notice**

This issue has been automatically marked as stale due to inactivity. It will be closed in **2 weeks** (October 30, 2025) if no further activity occurs.

If this issue is still relevant and you would like to keep it open, please:
- Comment on this issue to show continued interest
- Provide any additional information or updates
- Confirm that the issue still exists in the latest version

Thank you for your contribution to this project! 🙏
